crates/core/tedge/src/cli/diag/collect.rs

```diff
@@ -52,31 +55,48 @@ impl Command for DiagCollectCommand {
 
         let mut skipped_count = 0;
         let mut error_count = 0;
+        let mut log_buffer = Vec::new(); // To avoid output while spinner is running
```

Comment:
Why defer logging in `summary.log`? This is working fine without.

Reply:
Because it didn't work when I tried without the buffer and just call `diag_warning!()` or so on. The `summary.log` was created correctly. However, duplicated messages are printed in the terminal.